java/java.hints/src/org/netbeans/modules/java/hints/jdk/ConvertToNestedRecordPattern.java

```diff
@@ -0,0 +1,261 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.EnumSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.lang.model.SourceVersion;
+import javax.lang.model.element.Modifier;
+import javax.lang.model.element.RecordComponentElement;
+import javax.lang.model.element.TypeElement;
+import org.netbeans.api.java.source.CompilationInfo;
+import org.netbeans.api.java.source.TreePathHandle;
+import org.netbeans.api.java.source.CodeStyleUtils;
+import org.netbeans.api.java.source.WorkingCopy;
+import org.netbeans.api.java.queries.CompilerOptionsQuery;
+import org.netbeans.modules.java.hints.errors.Utilities;
+import org.netbeans.spi.editor.hints.ErrorDescription;
+import org.netbeans.spi.editor.hints.Fix;
+import org.netbeans.spi.java.hints.ErrorDescriptionFactory;
+import org.netbeans.spi.java.hints.Hint;
+import org.netbeans.spi.java.hints.HintContext;
+import org.netbeans.spi.java.hints.JavaFix;
+import org.netbeans.spi.java.hints.MatcherUtilities;
+import org.netbeans.spi.java.hints.TriggerTreeKind;
+import org.openide.util.NbBundle;
+
+@NbBundle.Messages({
+    "DN_ConvertToNestedRecordPattern=Convert to nested record pattern",
+    "DESC_ConvertToNestedRecordPattern=Convert to nested record pattern",
+    "ERR_ConvertToNestedRecordPattern=Convert to nested record pattern",
+    "FIX_ConvertToNestedRecordPattern=Convert to nested record pattern"
+})
+@Hint(displayName = "#DN_ConvertToNestedRecordPattern", description = "#DESC_ConvertToNestedRecordPattern", category = "rules15",
+        minSourceVersion = "19")
+/**
+ *
+ * @author mjayan
+ */
+public class ConvertToNestedRecordPattern {
+
+    private static final int RECORD_PATTERN_PREVIEW_JDK_VERSION = 19;
+
+    @TriggerTreeKind(Tree.Kind.DECONSTRUCTION_PATTERN)
+    public static ErrorDescription convertToNestedRecordPattern(HintContext ctx) {
+        if (Utilities.isJDKVersionLower(RECORD_PATTERN_PREVIEW_JDK_VERSION) && !CompilerOptionsQuery.getOptions(ctx.getInfo().getFileObject()).getArguments().contains("--enable-preview")) {
+            return null;
+        }
+        TreePath t = ctx.getPath();
+        if (!t.getParentPath().getLeaf().getKind().equals(Tree.Kind.INSTANCE_OF)) {
+            return null;
+        }
+        Set<String> recordPatternVarSet = new HashSet<>();
+        Map<PatternTree, List<PatternTree>> recordComponentMap = new LinkedHashMap<>();
+        DeconstructionPatternTree recordPattern = (DeconstructionPatternTree) t.getLeaf();
+        recordComponentMap = findNested(recordPattern, recordComponentMap);
+
+        for (PatternTree p : recordComponentMap.keySet()) {
+            BindingPatternTree bTree = (BindingPatternTree) p;
+            recordPatternVarSet.add(bTree.getVariable().getName().toString());
+        }
+        while (t != null && t.getLeaf().getKind() != Tree.Kind.IF) {
```

Comment:
Note this will find an `if` arbitrarily far from the original pattern. Even in the current code, it might be something like:
```
if (true) {
    boolean b = o instanceof R1(R2 r2) && r2.v() != null;
}
```

As I read the code, the above would probably work (more or less by accident), but would not without the enclosing `if`. Also, the scope of a variable may extend beyond the current statement, like in:
```
if (!(o instanceof R1(R2 r2))) { return ; }
System.err.println(r2.v()); //OK to use r2 here
}
```

Maybe we want to find the innermost enclosing block (or innermost enclosing lambda or case, as lambdas and cases not necessarily have a block - OTOH, I suspect searching bigger chunk of the code shouldn't have functional effect, so it might be OK to simply scan the innermost block for simplicity)?